lib/iris/_data_manager.py

```diff
@@ -390,7 +421,7 @@ def data(self, data):
         # Reset the fill-value appropriately.
         if ma.isMaskedArray(data):
             # Align the data manager fill-value with the numpy fill-value.
-            self.fill_value = data.fill_value
+            self._propagate_masked_data_fill_value()
```

Comment:
Right, so if you instantiate a `DataManager` with a real masked array this method will be called twice -- once here and then once again in the init (on [L80](https://github.com/bjlittle/iris/blob/638c0c665a3945cfeb72e8550a82c148f9a39138/lib/iris/_data_manager.py#L80)). I wonder if you could remove the second call?

* If you add that new method I asked for in a previous comment you could replicate the "[init_done](https://github.com/bjlittle/iris/blob/638c0c665a3945cfeb72e8550a82c148f9a39138/lib/iris/_data_manager.py#L395)" functionality from the data setter to make sure it's only run once.
* You could remove [the particular check](https://github.com/bjlittle/iris/blob/638c0c665a3945cfeb72e8550a82c148f9a39138/lib/iris/_data_manager.py#L77-L80) from the init knowing it's already been done here and only run [the second half](https://github.com/bjlittle/iris/blob/638c0c665a3945cfeb72e8550a82c148f9a39138/lib/iris/_data_manager.py#L81-L85) of the check. (A benefit of this approach would be that it would remove most of the newly-introduced code in the init, which would be pleasing.)
* Or something else of your devising.
* Or just not bother!